src/oemof/network/network/entity.py

```diff
@@ -51,7 +51,10 @@ def __init__(self, label, *, custom_properties=None):
         self.custom_properties = custom_properties
 
     def __eq__(self, other):
-        return id(self) == id(other)
+        if isinstance(other, Entity):
+            return self._label == other._label
+        else:
+            return self._label == other
 
     def __lt__(self, other):
         return str(self) < str(other)
```

Comment:
For consistency, it would make sense to change this as well.
```suggestion
    def __lt__(self, other):
        if isinstance(other, Entity):
            return self._label < other._label
        else:
            return self._label < other
```

(Will change results if different classes are compared.)